CenterForOpenScience/osf.io
# [ENG-2435] Upgrade pytest and others to fix broken tests

## Purpose

This is a booster shot of updated decencies for the OSF for pytest, these such keep us more up to date with the current test infrastructure and hopefully fix some probabilistic test fails.

You can read about the bug [here](https://github.com/pytest-dev/pytest/issues/3171) which indicates updating pytest will fix the issue. I picked pytest 5.0.0 because it is recent and only requires few changes. The biggest change is pytest no long allows passing of fixtures the same way. [Read about it here](https://github.com/pytest-dev/pytest/issues/3950). So that had to be changed to release this. 



## Changes

- update pytest and related dependencies
- turn fixtures into functions to appease pytest update
 


## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that travis tests pass as expected

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Documentation



## Side Effects

None that I know of. 

## Ticket

https://openscience.atlassian.net/browse/ENG-2435
